org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/IProblem.java

```diff
@@ -2545,4 +2545,9 @@ public interface IProblem {
 	 * @noreference preview feature
 	 */
 	int RawTypeInRecordPattern =  PreviewRelated + 1915;
+	/**
+	 * @since 3.33
+	 * @noreference preview feature
+	 */
+	int CannotInferRecordPatternTypes = PreviewRelated + 1940;
```

Comment:
commit 383db211d66ab0bfd3506b70baa47e830b10a627 from Jay has changes pertaining to this - since he was working on this, didn't add this.. Am rebasing to that commit and we should be having the jenkins runs.